Source/WTF/wtf/CallbackAggregator.h

```diff
@@ -32,19 +32,27 @@
 
 namespace WTF {
 
-class CallbackAggregator : public ThreadSafeRefCounted<CallbackAggregator> {
+template <DestructionThread destructionThread>
+class CallbackAggregatorOnThread : public ThreadSafeRefCounted<CallbackAggregatorOnThread<destructionThread>> {
 public:
-    static Ref<CallbackAggregator> create(CompletionHandler<void()>&& callback) { return adoptRef(*new CallbackAggregator(WTFMove(callback))); }
+    static auto create(CompletionHandler<void()>&& callback) { return adoptRef(*new CallbackAggregatorOnThread(WTFMove(callback))); }
 
-    ~CallbackAggregator()
+    ~CallbackAggregatorOnThread()
     {
-        ASSERT(m_wasConstructedOnMainThread == isMainThread());
-        if (m_callback)
-            m_callback();
+        if constexpr (destructionThread == DestructionThread::MainRunLoop) {
+            ensureOnMainRunLoop([callback = WTFMove(m_callback)]() mutable {
```

Comment:
This way, we wouldn't need logic like this. The destructor would get called on the thread that the client expected.

Reply:
Yeah, that makes sense. I didn't realize `ThreadSafeRefCounted<DestructionThread::MainRunLoop>` also ensured that the destructor would be invoked on the main runloop, despite the obvious name 🤦🏻‍♂️